framework/include/utils/SymmetricRankTwoTensor.h

```diff
@@ -0,0 +1,640 @@
+
+  /// returns -_vals
+  SymmetricRankTwoTensorTempl<T> operator-() const;
+
+  /// performs _vals *= a
+  SymmetricRankTwoTensorTempl<T> & operator*=(const T & a);
+
+  /// returns _vals*a
+  template <typename T2>
+  auto operator*(const T2 & a) const ->
+      typename std::enable_if<ScalarTraits<T2>::value,
+                              SymmetricRankTwoTensorTempl<decltype(T() * T2())>>::type;
+
+  /// performs _vals /= a
+  SymmetricRankTwoTensorTempl<T> & operator/=(const T & a);
+
+  /// returns _vals/a
+  template <typename T2>
+  auto operator/(const T2 & a) const ->
+      typename std::enable_if<ScalarTraits<T2>::value,
+                              SymmetricRankTwoTensorTempl<decltype(T() / T2())>>::type;
+
+  /// Defines multiplication with a vector to get a vector
+  template <typename T2>
+  TypeVector<typename CompareTypes<T, T2>::supertype> operator*(const TypeVector<T2> & a) const;
+
+  /// Defines logical equality with another SymmetricRankTwoTensorTempl<T2>
+  template <typename T2>
+  bool operator==(const SymmetricRankTwoTensorTempl<T2> & a) const;
+
+  /// Defines logical inequality with another SymmetricRankTwoTensorTempl<T2>
+  template <typename T2>
+  bool operator!=(const SymmetricRankTwoTensorTempl<T2> & a) const;
+
+  /// Sets _vals to the values in a ColumnMajorMatrix (must be 3x3)
+  SymmetricRankTwoTensorTempl<T> & operator=(const ColumnMajorMatrixTempl<T> & a);
+
+  /// returns _vals_ij * a_ij (sum on i, j)
+  T doubleContraction(const SymmetricRankTwoTensorTempl<T> & a) const;
+
+  /// returns C_ijkl = a_ij * b_kl
+  SymmetricRankFourTensorTempl<T> outerProduct(const SymmetricRankTwoTensorTempl<T> & a) const;
+
+  /// return positive projection tensor of eigen-decomposition
+  template <typename T2 = T>
+  typename std::enable_if<MooseUtils::IsLikeReal<T2>::value, SymmetricRankFourTensorTempl<T>>::type
+  positiveProjectionEigenDecomposition(std::vector<T> &, RankTwoTensorTempl<T> &) const;
+  template <typename T2 = T>
+  typename std::enable_if<!MooseUtils::IsLikeReal<T2>::value, SymmetricRankFourTensorTempl<T>>::type
+  positiveProjectionEigenDecomposition(std::vector<T> &, RankTwoTensorTempl<T> &) const;
+
+  /// returns A_ij - de_ij*tr(A)/3, where A are the _vals
+  SymmetricRankTwoTensorTempl<T> deviatoric() const;
+
+  /// returns the trace of the tensor, ie _vals[i][i] (sum i = 0, 1, 2)
+  T trace() const;
+
+  /// retuns the inverse of the tensor
+  SymmetricRankTwoTensorTempl<T> inverse() const;
+
+  /**
+   * Denote the _vals[i][j] by A_ij, then this returns
+   * d(trace)/dA_ij
+   */
+  SymmetricRankTwoTensorTempl<T> dtrace() const;
+
+  /**
+   * Denote the _vals[i][j] by A_ij, then
+   * S_ij = A_ij - de_ij*tr(A)/3
+   * Then this returns (S_ij + S_ji)*(S_ij + S_ji)/8
+   * Note the explicit symmeterisation
+   */
+  T generalSecondInvariant() const;
+
+  /**
+   * Calculates the second invariant (I2) of a tensor
+   */
+  T secondInvariant() const;
```

Comment:
The doxygen comments are reversed. The comment for `generalSecondInvariant` should be above `secondInvariant`, and vice versa.

Reply:
Same thing in `RankTwoTensor.h`

Reply:
The doxygen comments are still reversed.